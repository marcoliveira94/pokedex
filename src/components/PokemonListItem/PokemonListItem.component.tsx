import React from 'react'
import { Pokemon } from '../../models/pokemonList.type'
import styled from 'styled-components'
import { Text, Image, TouchableOpacity, Dimensions } from 'react-native'
import { useDominantColor } from '../../shared/hooks/useDominantColor.hook'

interface PokemonListItemProps {
  item: Pokemon
  navigate: () => void
}

const PokemonListItem = ({ item, navigate }: PokemonListItemProps) => {
  const urlImage = `https://pokeres.bastionbot.org/images/pokemon/${
    item.url.split('/')[6]
  }.png`
  const { color } = useDominantColor(urlImage, 'black')
  return (
    <ItemView background={color.background} onPress={navigate}>
      <ItemImage testID={'pokemon-list-image'} source={{ uri: urlImage }} />
      <ItemText>{item.name}</ItemText>
    </ItemView>
  )
}

export default PokemonListItem

const ItemView = styled(TouchableOpacity)<{ background: string }>`
  background-color: ${(props) => props.background};
  flex: 1;
  flex-direction: column;
  margin: 10px;
  height: ${Dimensions.get('window').width / 2 - 20}px;
  max-width: ${Dimensions.get('window').width / 2 - 20}px;
  border-radius: 20px;
  justify-content: center;
  align-items: center;
`

const ItemText = styled(Text)`
  font-size: 20px;
`

const ItemImage = styled(Image)`
  width: 50%;
  height: 50%;
`
