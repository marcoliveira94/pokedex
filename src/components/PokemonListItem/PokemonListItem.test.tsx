import React from 'react'
import PokemonListItem from './PokemonListItem.component'
import { render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

const navigate = jest.fn()
const mockedItem = {
  name: 'pokemonName',
  url: 'https://pokeapi.co/api/v2/pokemon/38/'
}

jest.mock('../../shared/hooks/useDominantColor.hook', () => ({
  useDominantColor: () => ({
    color: {
      background: 'black',
      detail: 'black',
      platform: undefined,
      primary: 'black',
      secondary: 'black'
    },
    error: null
  })
}))

describe('PokemonListItem', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PokemonListItem item={mockedItem} navigate={navigate} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders text correctly', () => {
    const element = render(
      <PokemonListItem item={mockedItem} navigate={navigate} />
    )
    const text = element.getByText(mockedItem.name)

    expect(text).toBeDefined()
  })

  it('renders image correctly', () => {
    const element = render(
      <PokemonListItem item={mockedItem} navigate={navigate} />
    )
    const image = element.getByTestId('pokemon-list-image')
    expect(image.props.source.uri).toBe(
      'https://pokeres.bastionbot.org/images/pokemon/38.png'
    )
  })
})
