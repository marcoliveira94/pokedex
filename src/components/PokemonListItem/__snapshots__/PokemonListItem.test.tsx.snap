// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonListItem renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Image
    className="sc-AxiKw huYQCN"
    source={
      Object {
        "uri": "https://pokeres.bastionbot.org/images/pokemon/38.png",
      }
    }
    testID="pokemon-list-image"
  />
  <Text
    className="sc-AxirZ cdGjCG"
  >
    pokemonName
  </Text>
</View>
`;
